app/src/main/java/com/example/sharewhatyoucanproject/MainActivity.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.sharewhatyoucanproject
+
+import android.os.Bundle
+import android.view.Menu
+import androidx.appcompat.app.AppCompatActivity
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.navigateUp
+import androidx.navigation.ui.setupActionBarWithNavController
+import androidx.navigation.ui.setupWithNavController
+import com.example.sharewhatyoucanproject.databinding.ActivityMainBinding
+import com.example.sharewhatyoucanproject.databinding.NavHeaderDrawerBinding
+import com.example.sharewhatyoucanproject.utils.createNotificationChannel
+import com.google.android.material.navigation.NavigationView
+import com.google.android.material.progressindicator.CircularProgressIndicator
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var appBarConfiguration: AppBarConfiguration
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var drawerLayout: DrawerLayout
+    lateinit var circularProgressIndicator: CircularProgressIndicator
+    private lateinit var mainViewModel: MainViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        mainViewModel = ViewModelProvider(this)[MainViewModel::class.java]
+        circularProgressIndicator = binding.progressCircular
+        setSupportActionBar(binding.appBarDrawer.toolbar)
+        drawerLayout = binding.drawerLayout
+        val navView: NavigationView = binding.navView
+        val navHostFragment =
+            supportFragmentManager.findFragmentById(R.id.nav_host_fragment_content_drawer) as NavHostFragment
+        val navController = navHostFragment.navController
+
+        appBarConfiguration = AppBarConfiguration(setOf(R.id.home_fragment), drawerLayout)
+        setupActionBarWithNavController(navController, appBarConfiguration)
+        navView.setupWithNavController(navController)
+        val headerView = navView.getHeaderView(0)
+        val headerBinding: NavHeaderDrawerBinding = NavHeaderDrawerBinding.bind(headerView)
+
+        mainViewModel.username.observe(this) { username ->
+            headerBinding.donertv.text = username
+        }
+        createNotificationChannel(this, mainViewModel.getUserId())
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.drawer, menu)
+        return true
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        val navController = findNavController(R.id.nav_host_fragment_content_drawer)
+        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
+    }
+
+    object DonorConstants {
+        const val CHANNEL_1_ID = "myCh"
```

Comment:
more descriptive constants will help other folks understand what these channels are for